src/unix/core.c

```diff
@@ -346,24 +346,50 @@ int uv_run(uv_loop_t* loop, uv_run_mode mode) {
   int timeout;
   int r;
   int ran_pending;
+  int emit_stats;
+  size_t count = 0;
+  size_t timers_count = 0;
 
   r = uv__loop_alive(loop);
   if (!r)
     uv__update_time(loop);
 
   while (r != 0 && loop->stop_flag == 0) {
+    emit_stats = loop->stats != NULL;
     uv__update_time(loop);
-    uv__run_timers(loop);
+    uv__update_stats_ts(loop, tick_start);
+  
+    uv__update_stats_ts(loop, timers1_start);
+    timers_count = uv__run_timers(loop);
+    uv__update_stats_ts(loop, timers1_end);
+
+    uv__update_stats_ts(loop, pending_start);
     ran_pending = uv__run_pending(loop);
-    uv__run_idle(loop);
-    uv__run_prepare(loop);
+    uv__update_stats_ts(loop, pending_end);
+
+    uv__update_stats_ts(loop, idle_start);
+    count = uv__run_idle(loop);
+    uv__update_stats_count(loop, idle_count, count);
+    uv__update_stats_ts(loop, idle_end);
+
+    uv__update_stats_ts(loop, prepare_start);
+    count = uv__run_prepare(loop);
+    uv__update_stats_count(loop, prepare_count, count);
+    uv__update_stats_ts(loop, prepare_end);
 
     timeout = 0;
     if ((mode == UV_RUN_ONCE && !ran_pending) || mode == UV_RUN_DEFAULT)
       timeout = uv_backend_timeout(loop);
 
+    uv__update_stats_ts(loop, poll_start);
```

Comment:
is there something that does not allow us to put a stats regarding the timeout used during the current tick?

Reply:
We certainly could add that... I'm curious about the use case for it but it would be trivial to add.

Reply:
In case the poll time is greater than timeout this might indicate that the CPU was too busy giving the CPU time to another process. So, the timeout could be used to apply something like that:
```
sleep_time = min(timeout, poll_time)
```

So, yes it's a nice to have, but having it at hand it's too kindy to don't ask for it :)